### PR TITLE
Mapped fields from Update Post endpoint to domain model using Automapper

### PR DESCRIPTION
**Pull request by en-aleksandar-stojmenovic**

### CODE REVIEW

File: ProjectSetup/Repositories/PostRepository.cs [hunk 1]
```diff
@@ -50,12 +49,9 @@ public async Task<Post> CreatePost(Post post)
 			return post;
 		}
 
-		public async Task<Post> UpdatePost(UpdatePostRequest postRequest)
+		public async Task<Post> UpdatePost(Post postRequest)
 		{
-			Post post = await FindByCondition(post => post.Id.Equals(postRequest.Id))
```
Comment (bronjaharis) [inline ProjectSetup/Repositories/PostRepository.cs:55]: Why did you remove this?

File: ProjectSetup/Controllers/V1/PostController.cs [hunk 2]
```diff
@@ -79,20 +79,24 @@ public async Task<IActionResult> DeletePostById([FromRoute] Guid postId)
...
-			var userOwnsPost = await _repository.Post.UserOwnsPostAsync(postRequest.Id, HttpContext.GetUserId());
+			var userOwnsPost = await _repository.Post.UserOwnsPostAsync(updateRequest.Id, HttpContext.GetUserId());
 
 			if (!userOwnsPost)
 			{
 				throw new PostBadRequestException("You don't own this post");
 			}
 
-			var post = await _repository.Post.UpdatePost(postRequest);
+			var post = await _repository.Post.FindPostByIdAsync(updateRequest.Id);
```
Comment (bronjaharis) [inline ProjectSetup/Controllers/V1/PostController.cs:91]: Do you check somewhere if post == null?

File: ProjectSetup/Repositories/PostRepository.cs (hunk 1 above)
Comment (en-aleksandar-stojmenovic) [reply ProjectSetup/Repositories/PostRepository.cs:55]: I removed this because I'm getting post by id in PostController

File: ProjectSetup/Controllers/V1/PostController.cs (hunk 2 above)
Comment (en-aleksandar-stojmenovic) [reply ProjectSetup/Controllers/V1/PostController.cs:91]: Yes, that check is in the method FindPostByIdAsync().

Comment (bronjaharis) [reply ProjectSetup/Controllers/V1/PostController.cs:91]: And will it throw an exception or will continue to the mapper in next line trying to map null?

Comment (en-aleksandar-stojmenovic) [reply ProjectSetup/Controllers/V1/PostController.cs:91]: Yes, it will throw PostNotFoundException

**Review by bronjaharis — APPROVED** [review]
